activerecord/test/cases/associations/nested_through_associations_test.rb

```diff
@@ -610,6 +610,12 @@ def test_polymorphic_has_many_through_joined_different_table_twice
     assert_equal hotel, Hotel.joins(:cake_designers, :drink_designers).take
   end
 
+  def test_has_many_through_reset_source_reflection_after_loading_is_complete
+    preloaded = Category.preload(:ordered_post_comments).find(1, 2).last
```

Comment:
Why not using just `.find(2)`?

Reply:
See my comment 2 lines below. This test is for a very tricky situation.